examples/metrics-basic/Cargo.toml

```diff
@@ -6,8 +6,12 @@ license = "Apache-2.0"
 publish = false
 
 [dependencies]
-opentelemetry = { path = "../../opentelemetry", features = ["metrics"] }
+opentelemetry = { path = "../../opentelemetry", features = ["metrics", "sync-gauge-experimental"] }
```

Comment:
maybe we should develop some convention saying, all experimental features must begin with "experimental-feature"?

Reply:
Yeah, good point. Prefixing `experimental` gives more clarity. I can make this `experimental-sync-gauge` if we agree.

Reply:
`experimental-sync-gauge` This looks good for me.

Reply:
> Should we consider an orthogonal unstable feature flag in general, which can be combined with other features which enable signals such as metrics, tracing, logs, rather than a unique feature flag name for all the combinations?

That would make things simpler, and easy to maintain. Just wondering if we need to take care of a scenario where the user wants to enable only a specific feature flag under (say) `metrics`. With generic flag like `unstable`, this mayn't be possible.

> Finally, how does this overlap with our previous decision about having opentelemetry crate be only for the stable API and the other opentelemetry-contrib crate being for the experimental or proposed, but not yet stable, additions?

Probably I missed the discussion. My proposal would be to have `opentelemetry-contrib` crate (under `otel-rust-contrib` repo) only for features that are not part of the specs. Convenience/wrapper API would be one such candidate. The main repo should have under feature flag
  - unstable implementation for stable/experimental specs feature
  - stable implementation of experimental specs feature.

From an end-user perspective, it would be easy to toggle the feature flag without changing dependencies. While we can debate whether this would increase the complexity of the code base without clear boundaries between stable and unstable implementation, but moving forward this seems manageable. Considering that all three signals – logs, traces, and metrics – are stable, significant changes in the specification aren't anticipated, making this strategy feasible.

Reply:
> The main repo should have under the feature flag:
> 
> unstable implementation for stable/experimental specs feature
> stable implementation of experimental specs feature.

Agree with this.

Reply:
I think we considered to use features for stable and non experimental feature to keep the compiling size minimal. Do we still see we need this going forward?

Reply:
Ok, as suggested above by @shaun-cox  and @jtescher, I have updated to use the conditional compilation of unstable/experimental features using the `otel_unstable` cfg flag - similar to how the `tokio` library using `tokio_unstable` for all unstable features.
So, to build the metrics with unstable features, we should be using:

```bash
$ RUSTFLAGS="--cfg=otel_unstable" cargo build --features "metrics"
```

Also updated the Contributing.md accordingly.

Reply:
could we add to the changelog too?

Reply:
> could we add to the changelog too?

done

Reply:
So would a good summary be:
For unstable/experimental features regardless of whether it's "experimental" in the spec or just "experimental" in our create we require the `otel_unstable` config flag.
For optional (non experimental) things we'll place them behind normal flags.

I would assume that certain things would maybe require both an `otel_unstable` config and a feature flag.

Reply:
> So would a good summary be:

Yes, rightly summarised -  `Configuration flag` for unstable features, and `feature flag` for optional features.

> Is there a reason we're choosing to use a configuration flag and not just a feature flag?

I felt using configuration flag for the unstable feature will keep our `Cargo.toml` clean, and avoid adding/enabling this flag in `Cargo.toml`  across multiple crates. Also, ease of testing, and writing examples - we can compile with or without feature with minimal changes. Also reduces the risk of inadvertently enabling this feature in Cargo.toml.

Reply:
Make sense, have changed it now to the feature flag `otel_unstable`, and updated the CONTRIBUTING.md accordingly.

Reply:
All okay with resolving this? 
https://github.com/open-telemetry/opentelemetry-rust/issues/1411 can be used for more feedback on this.